src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -632,6 +630,29 @@ def run_acr_build(self, dockerfile, source, quiet=False, build_from_source=False
                 dockerfile,
                 quiet,
             )
+            return False
+
+        # Only enable Cloud Build on Stage and Canary while the changes are deployed to all regions.
+        is_cloud_build_enabled = self.env.location and any(self.env.location.lower() == region for region in ["northcentralusstage", "centraluseuap", "eastus2euap"])
+        if (self.should_create_acr and is_cloud_build_enabled):
```

Comment:
ni: Remove redundant parentheses

Reply:
Thanks, updating this here and in other places.